File: algorithms/python/Challenge_Search_in_a_Binary_Search_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def searchBST(self, root: TreeNode, val: int) -> TreeNode:
        currRoot = root
        while currRoot:
            if val == currRoot.val:
                return currRoot
            elif val < currRoot.val:
                currRoot = currRoot.left
            elif val > currRoot.val:
                currRoot = currRoot.right
        
        return currRoot

File: algorithms/java/Jun.2020.challenge/Challenge_Search_in_a_Binary_Search_Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode searchBST(TreeNode root, int val) {
        TreeNode currRoot = root;
        
        while(currRoot != null) {
            if(val == currRoot.val)
                return currRoot;
            else if(val < currRoot.val)
                currRoot = currRoot.left;
            else
                currRoot = currRoot.right;
        }
        
        return currRoot;
    }
}
